File: java/po/world_simulator/entities/plants/Wolfberries.java
package po.world_simulator.entities.plants;

import po.world_simulator.entities.Entity;

import java.util.Map;

import po.world_simulator.Vector2;
import po.world_simulator.World;
import po.world_simulator.entities.Plant;

public class Wolfberries extends Plant {
    public Wolfberries(Vector2 position, int age) {
        super(99, position, "wolfberries", age);
    }

    public Wolfberries(Vector2 position) {
        this(position, 0);
    }

    public Wolfberries(Map<String, String> entityData){
        this(
            new Vector2(entityData.get("position")),
            Integer.parseInt(entityData.get("age"))
        );
    }

    @Override
    public void collision(Entity otherEntity) {
        otherEntity.kill();
        this.kill();
        World.getRenderer().addToLog(this.getName() + " tried to kill " + otherEntity.getName() + " and died");
    }
}

File: python/po/world_simulator/entities/plants/Wolfberries.py
from po.world_simulator.entities.Plant import Plant
from po.world_simulator.Vector2 import Vector2
import po.world_simulator.World as world_module

class Wolfberries(Plant):
    def __init__(self, position_or_data, age=0):
        if isinstance(position_or_data, dict):
            super().__init__(99, Vector2(position_or_data.get("position")), "wolfberries", int(position_or_data.get("age", 0)))
        else:
            super().__init__(99, position_or_data, "wolfberries", age)


    def collision(self, otherEntity):
        otherEntity.kill()
        self.kill()
        world_module.World.getRenderer().addToLog(self.getName() + " tried to kill " + otherEntity.getName() + " and died")
